server/storage/database_test.go

```diff
@@ -2,290 +2,330 @@ package storage
+
+	gorm1 := SampleTUF(0)
+	gorm1.ID = 1
+	gorm2 := GormTUFFile{ID: 2, Gun: "testGUN", Role: "targets", Version: 1,
+		Data: []byte("2")}
+	gorm3 := SampleTUF(2)
+	gorm3.ID = 3
+	expected := []GormTUFFile{gorm1, gorm2, gorm3}
+
+	var rows []GormTUFFile
+	query := gormDB.Find(&rows)
+	assert.NoError(t, query.Error)
+	assert.Equal(t, expected, rows)
+
+	dbStore.DB.Close()
+}
+
+// TestMySQLUpdateManyVersionOrder asserts that inserting updates with
+// non-monotonic versions still succeeds.
+func TestMySQLUpdateManyVersionOrder(t *testing.T) {
+	gormDB, dbStore := SetUpSQLite(t)
+
+	err := dbStore.UpdateMany(
+		"testGUN", []MetaUpdate{SampleUpdate(2), SampleUpdate(0)})
+	assert.NoError(t, err)
+
+	// the whole transaction should have rolled back, so there should be
+	// no entries.
+	gorm1 := SampleTUF(2)
+	gorm1.ID = 1
+	gorm2 := SampleTUF(0)
+	gorm2.ID = 2
+
+	var rows []GormTUFFile
+	query := gormDB.Find(&rows)
+	assert.NoError(t, query.Error)
+	assert.Equal(t, []GormTUFFile{gorm1, gorm2}, rows)
+
+	dbStore.DB.Close()
+}
+
+// TestMySQLUpdateManyDuplicateRollback asserts that inserting duplicate
+// updates fails.
+func TestMySQLUpdateManyDuplicateRollback(t *testing.T) {
+	gormDB, dbStore := SetUpSQLite(t)
+
+	update := SampleUpdate(0)
+	err := dbStore.UpdateMany("testGUN", []MetaUpdate{update, update})
+	assert.Error(t, err, "There should be an error updating twice.")
+	// sqlite3 error and mysql error aren't compatible
+	// assert.IsType(t, &ErrOldVersion{}, err,
+	//               "UpdateMany returned wrong error type")
+
+	// the whole transaction should have rolled back, so there should be
+	// no entries.
+	var count int
+	query := gormDB.Model(&GormTUFFile{}).Count(&count)
+	assert.NoError(t, query.Error)
+	assert.Equal(t, 0, count)
+
+	dbStore.DB.Close()
+}
+
+func TestMySQLGetCurrent(t *testing.T) {
+	gormDB, dbStore := SetUpSQLite(t)
+
+	byt, err := dbStore.GetCurrent("testGUN", "root")
+	assert.Nil(t, byt)
+	assert.Error(t, err, "There should be an error Getting an empty table")
+	assert.IsType(t, &ErrNotFound{}, err, "Should get a not found error")
 
-	sqlmock.ExpectQuery(
-		"SELECT `cipher`, `public` FROM `timestamp_keys` WHERE `gun`=\\?;",
-	).WithArgs("testGUN").WillReturnError(sql.ErrNoRows)
+	tuf := SampleTUF(0)
+	query := gormDB.Create(&tuf)
+	assert.NoError(t, query.Error, "Creating a row in an empty DB failed.")
 
-	_, _, err = s.GetTimestampKey("testGUN")
-	assert.IsType(t, &ErrNoKey{}, err, "Expected ErrNoKey from GetTimestampKey")
+	byt, err = dbStore.GetCurrent("testGUN", "root")
+	assert.NoError(t, err, "There should not be any errors getting.")
+	assert.Equal(t, []byte("1"), byt, "Returned data was incorrect")
 
-	//err = db.Close()
-	//assert.Nil(t, err, "Expectation not met: %v", err)
+	dbStore.DB.Close()
+}
+
+func TestMySQLDelete(t *testing.T) {
+	gormDB, dbStore := SetUpSQLite(t)
+
+	// Not testing deleting from an empty table, because that's not an error
```

Comment:
I plan on adding this test back in when the `database.go` implementation has been converted, because we can check the number of rows affected to see if the delete actually happened.